pkg/shared/util/grpc_utils.go

```diff
@@ -78,6 +151,13 @@ func WaitForServerInfo(timeout time.Duration, filePath string) (*info.ServerInfo
 		return nil, fmt.Errorf("failed to read server info: %w", err)
 	}
 
+	// If we are testing locally or in CI, we skip checking for compatibility issues
+	if strings.Contains(numaflow.GetVersion().Version, "latest") || os.Getenv("GITHUB_ACTIONS") == "true" {
+		return serverInfo, nil
```

Comment:
In this case, checking if the sdk version meets the min requirement still applies.